src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -308,7 +308,10 @@ class AppsController(
         val healthStatuses = await(healthCheckManager.statuses(appId))
         instances.flatMap { instance =>
           val health = healthStatuses.getOrElse(instance.instanceId, Nil)
-          instance.tasksMap.values.map { task => EnrichedTask(appId, task, instance.agentInfo, health) }
+          instance.tasksMap.values.map { task =>
+            EnrichedTask(
```

Comment:
what do you think about factory method for EnrichedTask that would take instance? If there is possibility that even more properties from instance will be read like this, it might make sense...

Reply:
👍 second this

Reply:
👍 second this

Reply:
Agreed.